server/src/repositories/person.repository.ts

```diff
@@ -182,6 +197,29 @@ export class PersonRepository implements IPersonRepository {
     return queryBuilder.getMany();
   }
 
+  @GenerateSql({ params: [DummyValue.UUID, [DummyValue.STRING], { withHidden: true }] })
+  getManyByName(
+    userId: string,
+    personNames: string[],
+    { withHidden }: PersonNameSearchOptions,
+  ): Promise<PersonEntity[]> {
+    const queryBuilder = this.personRepository
+      .createQueryBuilder('person')
+      .leftJoin('person.faces', 'face')
+      .where('person.ownerId = :userId AND (LOWER(person.name) IN (:...names))', {
+        userId,
+        names: personNames.map((n) => n.toLowerCase()),
+      })
+      .groupBy('person.id')
+      .orderBy('COUNT(face.assetId)', 'DESC')
```

Comment:
Counting = scanning the entire `asset_faces` table, which can have hundreds of thousands of rows. Doing that for every asset is not a good idea. I would remove the left join, ordering, limit and group by.

Reply:
Done: https://github.com/immich-app/immich/compare/main...bugfest:immich:faces-from-metadata#diff-bdd10fc89e503edc0aa770142c760f507c6b6817b12b1b3a28df3e7d7145ca69R206-R211